packages/react/src/provider.tsx

```diff
@@ -0,0 +1,87 @@
+import LogtoClient, { LogtoConfig } from '@logto/browser';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+import { LogtoContext, ExtendedLogtoContextProps } from './context';
+
+export type LogtoProviderProps = {
+  logtoConfig: LogtoConfig;
+  children?: React.ReactNode;
+};
+
+export const LogtoProvider = ({ logtoConfig, children }: LogtoProviderProps) => {
+  const logtoClientReference = useRef(new LogtoClient(logtoConfig));
+  const [logtoContext, setLogtoContext] = useState<ExtendedLogtoContextProps>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const signIn = useCallback(async (redirectUri: string) => {
+    setIsLoading(true);
+    await logtoClientReference.current.signIn(redirectUri);
+    setIsLoading(false);
+  }, []);
+
+  const handleSignInCallback = useCallback(async (callbackUri: string) => {
+    setIsLoading(true);
+    await logtoClientReference.current.handleSignInCallback(callbackUri);
+    setIsLoading(false);
+  }, []);
+
+  const signOut = useCallback(async (postLogoutRedirectUri: string) => {
+    setIsLoading(true);
+    await logtoClientReference.current.signOut(postLogoutRedirectUri);
+    setIsLoading(false);
+  }, []);
+
+  const fetchUserInfo = useCallback(async () => {
+    setIsLoading(true);
+    const userInfo = await logtoClientReference.current.fetchUserInfo();
+    setIsLoading(false);
+
+    return userInfo;
+  }, []);
+
+  const getAccessToken = useCallback(async () => {
+    setIsLoading(true);
+    const accessToken = await logtoClientReference.current.getAccessToken();
+    setIsLoading(false);
+
+    return accessToken;
+  }, []);
+
+  const getIdTokenClaims = useCallback(() => {
+    return logtoClientReference.current.getIdTokenClaims();
+  }, []);
+
+  useEffect(() => {
+    setLogtoContext({
+      logtoClient: logtoClientReference.current,
+      isAuthenticated: logtoClientReference.current.isAuthenticated,
```

Comment:
looks like this won't be updated in time? there's no dependency in the hook. better expose a function?

Reply:
Updated, good call